modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -23,6 +37,317 @@
+            for cols in col_partitions:
+                all_partitions[-1].append(
+                    cls.deploy(
+                        cls.parse_fsspec_files,
+                        files_for_parser=files_to_read,
+                        columns=cols,
+                        num_returns=3,
+                        storage_options=storage_options,
+                        **kwargs,
+                    )
+                )
+        return all_partitions
+
+    @classmethod
+    def build_partition(cls, partition_ids, column_widths):
+        """
+        Build array with partitions of `cls.frame_partition_cls` class.
+
+        Parameters
+        ----------
+        partition_ids : list
+            Array with references to the partitions data.
+        column_widths : list
+            Number of columns in each partition.
+
+        Returns
+        -------
+        np.ndarray
+            array with shape equals to the shape of `partition_ids` and
+            filed with partition objects.
+        """
+        return np.array(
+            [
+                [
+                    cls.frame_partition_cls(
+                        partition_ids[i][j][0],
+                        length=partition_ids[i][j][2],
+                        width=column_widths[j],
+                    )
+                    for j in range(len(partition_ids[i]))
+                ]
+                for i in range(len(partition_ids))
+            ]
+        )
+
+    @classmethod
+    def build_index(cls, path, storage_options):
+        """
+        Compute index and its split sizes of resulting Modin DataFrame.
+
+        Parameters
+        ----------
+        path : Pathlike
+            Path to dataset.
+        storage_options : dict
+            Paramters for specific storage engine.
+
+        Returns
+        -------
+        index : pandas.Index
+            Index of resulting Modin DataFrame.
+        needs_index_sync : bool
+            Whether the partition indices need to be synced with frame
+            index because there's no index column, or at least one
+            index column is a RangeIndex.
+        """
+        fs, path_ = (
+            fsspec.core.url_to_fs(path, **storage_options)
+            if is_fsspec_url(path)
+            else (None, path)
+        )
+        pandas_metadata = ParquetDataset(
+            path_, filesystem=fs, use_legacy_dataset=False
+        ).schema.pandas_metadata
+        index_columns = []
+        if pandas_metadata is not None:
+            index_columns = pandas_metadata.get("index_columns", index_columns)
+        column_names_to_read = []
+        for column in index_columns:
+            if isinstance(column, str):
+                column_names_to_read.append(column)
+            elif column["name"] is not None:
+                column_names_to_read.append(column["name"])
+        complete_index = (
+            read_table(path, columns=column_names_to_read).to_pandas().index
+        )
```

Comment:
No, this will not read the entire dataset. This will only read in columns that are index columns. There are cases where parquet will have a separate column that is the index column I believe.

Reply:
@vnlitvinov the `read_table` here is from pyarrow - so presumably pyarrow can do an efficient read of just the columns we want, and then the `to_pandas` converts it from a pyarrow table to a pandas df, and we grab the index from there.